src/App/analytics.ts

```diff
@@ -0,0 +1,61 @@
+import GoogleAnalytics, { resetClientId } from 'electron-ga-uuid';
+import { reaction } from 'mobx';
+import { AnalyticsInterface } from '../../public/models/analytic.model';
+import isDev from 'electron-is-dev';
+
+const DEV_TRACK_ID = 'UA-169320344-1';
+const TRACK_ID = 'UA-22170471-17';
+
+export default class Analytics implements AnalyticsInterface {
+  constructor(settings: { enableAnalytics: boolean }) {
+    this.isEnabled = !isDev && settings.enableAnalytics;
+    this.ga = null;
+
+    const updateProxy = (enableAnalytics: boolean): void => {
+      this.isEnabled = !isDev && enableAnalytics;
+      if (this.isEnabled) this.ga = new GoogleAnalytics(isDev ? DEV_TRACK_ID : TRACK_ID);
+    };
+    updateProxy(settings.enableAnalytics);
+    reaction(
+      (): boolean => settings.enableAnalytics,
+      (enableAnalytics: boolean): void => {
+        updateProxy(enableAnalytics);
+      }
+    );
+  }
+  isEnabled: boolean;
+  ga: GoogleAnalytics | null;
```

Comment:
You can make this property optional and remove the `null`.

Reply:
I don't think you can make a class member/property optional (interfaces can have optional properties).  I simply removed the `null` and removed the assignment to null above.  My IDE is happy with it.

Reply:
it turns out while you cannot have an optional class property, you are still required to initialize it.  So we initialize it via shorthand to `undefined`